server/services/lists/caseloadService.ts

```diff
@@ -1,128 +1,144 @@
+          probationPractitioner: getProbationPractioner(coms, licence.comUsername, deliusRecord),
         }
       })
-      .filter(offender => offender.nomisRecord, 'unable to find prison record')
+      .filter(licence => licence.nomisRecord)
   }
 
-  private mapLicencesToOffenders = async (licences: LicenceSummary[], user?: User): Promise<ManagedCase[]> => {
-    const nomisIds = licences.map(l => l.nomisId)
-    const deliusRecords = await this.communityService.getOffendersByNomsNumbers(nomisIds)
-    const offenders = await this.pairDeliusRecordsWithNomis(deliusRecords, user)
-    return offenders.map(offender => {
+  private buildCase = async (licences: LicenceSummary[], user?: User): Promise<VaryApprovalCase[]> => {
+    assertContainsNoDuplicates(licences, l => l.nomisId)
+    const cases = await this.linkRecords(licences, user)
+
+    return cases.map(({ licence, deliusRecord, nomisRecord, probationPractitioner }) => {
+      const releaseDate = nomisRecord.releaseDate ? format(parseIsoDate(nomisRecord.releaseDate), 'dd MMM yyyy') : null
+
+      const variationRequestDate = licence.dateCreated
+        ? format(parseCvlDateTime(licence.dateCreated, { withSeconds: false }), 'dd MMMM yyyy')
+        : null
+
       return {
-        ...offender,
-        licences: licences
-          .filter(l => l.nomisId === offender.nomisRecord.prisonerNumber)
-          .map(l => {
-            const releaseDate = l.actualReleaseDate || l.conditionalReleaseDate
-            return {
-              id: l.licenceId,
-              type: <LicenceType>l.licenceType,
-              status: <LicenceStatus>l.licenceStatus,
-              comUsername: l.comUsername,
-              dateCreated: l.dateCreated,
-              approvedBy: l.approvedByName,
-              approvedDate: l.approvedDate,
-              versionOf: l.versionOf,
-              kind: <LicenceKind>l.kind,
-              licenceStartDate: l.licenceStartDate,
-              hardStopDate: parseCvlDate(l.hardStopDate),
-              hardStopWarningDate: parseCvlDate(l.hardStopWarningDate),
-              isDueToBeReleasedInTheNextTwoWorkingDays: l.isDueToBeReleasedInTheNextTwoWorkingDays,
-              updatedByFullName: l.updatedByFullName,
-              releaseDate: releaseDate ? parseCvlDate(releaseDate) : null,
-            }
-          }),
+        licenceId: licence.licenceId,
+        name: convertToTitleCase(`${nomisRecord.firstName} ${nomisRecord.lastName}`.trim()),
+        crnNumber: deliusRecord.otherIds.crn,
+        licenceType: <LicenceType>licence.licenceType,
+        variationRequestDate,
+        releaseDate,
+        probationPractitioner,
       }
     })
   }
 
-  private async mapResponsibleComsToCases(caseload: ManagedCase[]): Promise<ManagedCase[]> {
-    const comUsernames = caseload
-      .map(
-        offender =>
-          offender.licences.find(l => offender.licences.length === 1 || l.status !== LicenceStatus.ACTIVE).comUsername
```

Comment:
The queries are always limited to return VARIATION_SUBMITTED so no need to check status.

There should only ever be one of these per noms id so I've created an assertion that ensures there's no duplicates.